### PR TITLE
build: update to axe v4.2

### PR DESCRIPTION
**Pull request by mohanraj-r**
### Build 🏗️ 
* Update to axe v4.2.3 
  - without adding the new rules (will be tackled in a future PR)
  - this brings in dozens of bug fixes in recent axe release
  - update test snapshots to reflect version changes in axe help URL 
  - update tests to account for removed rule (`bypass`)

### CODE REVIEW

**Comment by mohanraj-r** [comment]
Will investigate the codecov drop..

**Comment by mohanraj-r** [comment]
Had to revert back to Jest v26 from attempted upgrade to v27 due to inconsistencies in code cov reports as noted in #65

File: packages/jest/__tests__/matcher.test.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ describe('toBeAccessible jest a11y matcher', () => {
 
     it.each(a11yConfigParams)('should throw error for dom with a11y issues with config: %#', async (config) => {
         document.body.innerHTML = domWithA11yIssues;
-        expect.assertions(a11yIssuesCount);
+        expect.assertions(a11yIssuesCount + 1);
```
Comment (trevor-bliss) [inline packages/jest/__tests__/matcher.test.ts:49]: why the +1 here now?

**Review by trevor-bliss — DISMISSED** [review]

File: packages/jest/__tests__/matcher.test.ts (hunk 1 above)
Comment (mohanraj-r) [reply packages/jest/__tests__/matcher.test.ts]: Good catch :) 
The test has 6 assertions, and a11yIssuesCount was changed from 6 to 5 in this PR to account for the removed `bypass` rule. But this test doesn't actually check for num of a11y issues - think it was a coincidence that the num of issues matched the num of assertions. 

Removed the expectation and added a `hasAssertions` check to the utils method that is using the async calls.

**Review by trevor-bliss — APPROVED** [review]

**Comment by mohanraj-r** [comment]
@trevor-bliss Can you please review the latest changes. I accidentally dismissed your approval.

**Review by trevor-bliss — APPROVED** [review]
Let's make sure to follow up to understand why the a11y checks don't work with Jest's fake timers.

**Comment by mohanraj-r** [comment]
Opened issue in axe-core https://github.com/dequelabs/axe-core/issues/3055

**Pull request merged**